tee-worker/local-setup/launch.py

```diff
@@ -58,19 +61,36 @@ def main(processes, config_path, parachain_type):
         print('Starting litentry-parachain done')
         print('----------------------------------------')
 
-    i = 1
+    c = pycurl.Curl()
+    c.setopt(pycurl.URL, 'http://localhost:4545/is_initialized')
+
+    worker_i = 0
+    worker_num = len(config["workers"])
     for w_conf in config["workers"]:
-        processes.append(run_worker(w_conf, i))
-        # sleep to prevent nonce clash when bootstrapping the enclave's account
-        sleep(3)
-        if i == 1:
-             # Give worker 1 some time to register itself, otherwise key & state sharing will not work.
-             #
-             # litentry: increase the gap between worker launch
-             #           we need a cleaner solution though, see https://github.com/integritee-network/worker/issues/731
-            sleep(180)
-
-        i += 1
+        processes.append(run_worker(w_conf, worker_i))
+        print()
+        if worker_i < worker_num:
+            counter = 0
+            while True:
+                sleep(5)
+                buffer = BytesIO()
+                c.setopt(c.WRITEDATA, buffer)
+                try:
+                    c.perform()
+                except:
+                    return 0
+
+                if "I am initialized." == buffer.getvalue().decode('iso-8859-1'):
+                    break
+                if counter >= 60:
+                    print("Worker initialization error. Exit")
+                    return 0
+                counter = counter + 1
+
+        worker_i += 1
+
+    c.close()
+    print("Worker(s) started!")
```

Comment:
I think you need to print "workers started" until the last worker is started (currently it looks for `localhost:4545` which is the first worker)

So when I see this message, worker2 is not fully started, actually

Reply:
Fixed.